src/Analyzer/FileParserFactory.php

```diff
@@ -9,13 +9,15 @@
 
 class FileParserFactory
 {
-    public static function createFileParser(TargetPhpVersion $targetPhpVersion, bool $parseCustomAnnotations = true): FileParser
-    {
+    public static function createFileParser(
+        TargetPhpVersion $targetPhpVersion = null,
```

Comment:
```suggestion
        ?TargetPhpVersion $targetPhpVersion = null,
```

I assume this should be explicitly marked as a nullable type instead of depending on implicit PHP behaviour.